tests/test_xbud_calendar.py

```diff
@@ -0,0 +1,169 @@
+            # 2nd Day of Christmas on Saturday, Dec 26
+            #   Note: 25th, 24th both holidays
+            pd.Timestamp('2015-12-23', tz=UTC),
+            pd.Timestamp('2015-12-28', tz=UTC),
+            # New Year's Eve on Saturday, Dec 31
+            pd.Timestamp('2016-12-30', tz=UTC),
+            pd.Timestamp('2017-01-02', tz=UTC),
+        ]
+
+        for session_label in expected_sessions:
+            self.assertIn(session_label, all_sessions)
+
+    def test_four_day_weekends(self):
+        all_sessions = self.calendar.all_sessions
+
+        # For most holidays, falling on a Tuesday or Thursday adds
+        # a holiday on Monday or Friday, respectively, creating a four
+        # day weekend.  This test makes sure the extra day gets added
+        expected_holidays = [
+            # New Years Day on Thursday, Jan 1
+            pd.Timestamp('2015-01-01', tz=UTC),
+            pd.Timestamp('2015-01-02', tz=UTC),
+            # National Holiday on Tuesday, March 15
+            pd.Timestamp('2016-03-15', tz=UTC),
+            pd.Timestamp('2016-03-14', tz=UTC),
+            # Labour Day on Tuesday, May 1
+            pd.Timestamp('2018-05-01', tz=UTC),
+            pd.Timestamp('2018-04-30', tz=UTC),
+            # St. Stephen's Day on Thursday, Aug 20
+            pd.Timestamp('2015-08-20', tz=UTC),
+            pd.Timestamp('2015-08-21', tz=UTC),
+            # National Holiday on Thursday, Oct 23
+            pd.Timestamp('2014-10-23', tz=UTC),
+            pd.Timestamp('2014-10-24', tz=UTC),
+            # All Saint's Day on Tuesday, Nov 1
+            pd.Timestamp('2016-11-01', tz=UTC),
+            pd.Timestamp('2016-10-31', tz=UTC),
+            # Second Day of Christmas on Thursday, Dec 26
+            pd.Timestamp('2019-12-26', tz=UTC),
+            pd.Timestamp('2019-12-27', tz=UTC),
+        ]
+
+        # Christmas Eve and New Years Eve do not use four day weekend
+        # rule.  This test makes sure the Monday before each is a
+        # trading day.
+        expected_sessions = [
+            # Dec 23 Mondays
+            pd.Timestamp('2002-12-23', tz=UTC),
+            pd.Timestamp('2013-12-23', tz=UTC),
+            pd.Timestamp('2019-12-23', tz=UTC),
+            # Dec 30 Mondays
+            pd.Timestamp('2002-12-30', tz=UTC),
+            pd.Timestamp('2013-12-30', tz=UTC),
+            pd.Timestamp('2019-12-30', tz=UTC),
+        ]
+
+        for holiday_label in expected_holidays:
+            self.assertNotIn(holiday_label, all_sessions)
+
+        for session_label in expected_sessions:
+            self.assertIn(session_label, all_sessions)
+
+    def test_new_years_eve(self):
+        """
+        New Year's Eve became a holiday in 2011.  Before that it was not
+        a holiday unless it fell on a Monday (because of New Year's Day
+        four day weekend rule)
+        """
+        all_sessions = self.calendar.all_sessions
+        session_days = (1, 2, 3, 4)
```

Comment:
In `trading_calendar.py` we define these days of weeks as constants (i.e. `TUESDAY=1`). That would be a little clearer to use here.

Reply:
Oh right, I must've forgotten about that.  And oops forgot to update the README, doing that now (and will include the 1.9s)!